tcp-echo.go

```diff
@@ -45,6 +52,44 @@ func main() {
 		message = message + fmt.Sprintf("Service %s.\n", serviceAccountName)
 	}
 
+	if tlsPort != "" && tlsCertFile != "" && tlsKeyFile != "" {
+
+		roots := x509.NewCertPool()
+		if tlsCACertFile != "" {
+			caCertPEM, err := os.ReadFile(tlsCACertFile)
+			if err != nil {
+				log.Panicf("failed to load CA, err %v", err)
+			}
+			roots.AppendCertsFromPEM(caCertPEM)
+		}
+
+		cert, err := tls.LoadX509KeyPair(tlsCertFile, tlsKeyFile)
+		if err != nil {
+			log.Panicf("failed to load cert, err %v", err)
+		}
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      roots,
+		}
+		tlsListener, err := tls.Listen("tcp", ":"+tlsPort, tlsConfig)
+		if err != nil {
+			log.Panicf("failed to listen on TLS port %s: %v", tlsPort, err)
+		}
+
+		log.Printf("listen TLS on port %s", tlsPort)
+		go func() {
+			for {
+				conn, err := tlsListener.Accept()
+				if err != nil {
+					log.Panicln("tls accept fail", err)
+				}
+
+				tlsMsg := message + "Through TLS connection.\n"
+				go handleTCPRequest(conn, tlsMsg)
+			}
+		}()
+	}
```

Comment:
If I read this right then  this will spawn a separate listener in a separate goroutine **additionally** to the non-tls listener. Would that be what we want? Wouldn't we want to spawn one or the other?

Secondly: I suggest to put this into a separate func and spawn either this of the non-tls listener, this way IMO it would be way cleaner.

WDYT?

Reply:
For the first question, I wanted to spawn the TLS listener and keep the non-TLS listener in the main goroutine because I wanted to always run the non-TLS listner. 
For the second, will do.

Reply:
I'd put it into a separate func anyway.

As for running both: what if someone wants to use only one port? Is there a use case where we need both?